adagrams/game.py

```diff
@@ -1,11 +1,123 @@
+import random
+
+LETTER_POOL = {
+    'A': 9, 
+    'B': 2, 
+    'C': 2, 
+    'D': 4, 
+    'E': 12, 
+    'F': 2, 
+    'G': 3, 
+    'H': 2, 
+    'I': 9, 
+    'J': 1, 
+    'K': 1, 
+    'L': 4, 
+    'M': 2, 
+    'N': 6, 
+    'O': 8, 
+    'P': 2, 
+    'Q': 1, 
+    'R': 6, 
+    'S': 4, 
+    'T': 6, 
+    'U': 4, 
+    'V': 2, 
+    'W': 2, 
+    'X': 1, 
+    'Y': 2, 
+    'Z': 1
+}
+
+
 def draw_letters():
-    pass
+    hand = []
+    copy_letter_pool = LETTER_POOL.copy()
+    while len(hand) < 10:
+      random_letter = random.choices(list(copy_letter_pool), weights = copy_letter_pool.values(), k=1)
+      str_random_letter = random_letter[0]
+      
+      if copy_letter_pool[str_random_letter] > 0:
+        hand.append(str_random_letter)
+        copy_letter_pool[str_random_letter] += -1
+    return hand
 
 def uses_available_letters(word, letter_bank):
-    pass
+    quantity_available = {}
+    validate_enough_available = {}
+    for letter in letter_bank:
+        available_count = letter_bank.count(letter)
+        quantity_available[letter] = available_count
 
+    for character in word.upper():
+        amount_needed_for_word = (word.upper()).count(character)
+        if character in quantity_available and (amount_needed_for_word <= quantity_available[character]):
+            validate_enough_available[character] = True
+        else:
+            validate_enough_available[character] = False
+  
+    if False in validate_enough_available.values():
+        return False
+    else:
+        return True
+    
```

Comment:
It took me a second, but this is a very unique way to write this function! If you wanted to cut this down just a bit, you could make a copy of your letter_bank. From there, you can loop through your word and check to see if each letter is in the copy of the letter_bank. If it is, you can remove it from the letter bank and if it isn't you can immediately return False and terminate the loop and function early! Any chance to terminate a function early is a great idea!